smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/model/ShapeExt.kt

```diff
@@ -108,14 +108,21 @@ fun ServiceShape.hasIdempotentTokenMember(model: Model): Boolean {
 /**
  * Return the formatted (Kotlin) function signature for the given operation
  */
-fun OperationIndex.operationSignature(model: Model, symbolProvider: SymbolProvider, op: OperationShape): String {
+fun OperationIndex.operationSignature(
+    model: Model,
+    symbolProvider: SymbolProvider,
+    op: OperationShape,
+    includeOptionalDefault: Boolean = false
```

Comment:
question

Why do we need this parameter? It seems like for any operation we should have only one possible signature.

Reply:
It has to do with where you're rendering the signature.

Kotlin requires default args to be defined at the superclass / interface level. So for us that's `{Service}Client`. We're not allowed to mirror that same default arg declaration at the override level ie. `Default{Service}Client`. Thus the parameter is introduced here to control that behavior.

I'm not really a diehard fan of flat-out bool args, the alternative in my mind would be to have the "create signature" method return some structure that the caller can then use conditionally (ie. `render()` vs `renderWithDefaults()`), but the effort vs. value added seemed not worth it just to avoid the bool in the one scenario this is used.

Reply:
Oh right, gotcha.

Boolean params are fine, especially when combined with named arguments at the call site:

```kotlin
opIndex.operationSignature(model, symbolProvider, op, includeOptionalDefault = true)
```

You might update the function documentation though to document that param in particular.

Reply:
If you really don't like boolean here (I don't care too much) you could switch to an enum that indicates the use case:

```kotlin
enum ParameterLocation {
    Definition,
    Override,
}
```